lib/Driver/Driver.cpp

```diff
@@ -2031,6 +2031,12 @@ const ToolChain *Driver::getToolChain(const ArgList &Args) const {
       TC = new toolchains::Darwin(*this, Target);
       break;
     case llvm::Triple::Linux:
+      if (Target.isAndroid()) {
```

Comment:
`toolchains::Windows` came from the Cygwin port, so if you rename it `toolchains::Cygwin`, that's a little less strange to inherit from `toolchains::GenericUnix`.